common/src/app/common/file_builder.cljc

```diff
@@ -561,35 +565,38 @@
           :id id}))))
 
 (defn start-component
-  [file data]
-
-  (let [selrect cts/empty-selrect
-        name               (:name data)
-        path               (:path data)
-        main-instance-id   (:main-instance-id data)
-        main-instance-page (:main-instance-page data)
-        obj (-> (cts/make-minimal-group nil selrect name)
-                (merge data)
-                (dissoc :path
-                        :main-instance-id
-                        :main-instance-page
-                        :main-instance-x
-                        :main-instance-y)
-                (check-name file :group)
-                (d/without-nils))]
-    (-> file
-        (commit-change
-         {:type :add-component
-          :id (:id obj)
-          :name name
-          :path path
-          :main-instance-id main-instance-id
-          :main-instance-page main-instance-page
-          :shapes [obj]})
-
-        (assoc :last-id (:id obj))
-        (update :parent-stack conjv (:id obj))
-        (assoc :current-component-id (:id obj)))))
+  ([file data] (start-component file data :group))
+  ([file data root-type]
+   (let [selrect (if (and (:x data) (:y data) (:width data) (:height data))
```

Comment:
You could use `(or (gsh/make-selrect ...) cts/empty-selrect)` because `make-selrect` returns null if one of its arguments is not a number

Reply:
done